### DISCUSSION TITLE
How to specify an IP address or a FQDN

### DISCUSSION THREAD

**Discussion by fcbry** [stefanprodan/timoni#420] [Q&A] [answered]
In timoni (and timoni only)

If I specify in my templates/config.cue :

```
  targetServer!: (net.FQDN| net.IP)
```

I can supply either a FQDN (i.e "cnn.com") or IP (i.e. "1.1.1.1" ) and it will validate. 

But if I specify: 

```
  targetServer!: ( net.IP| net.FQDN )
```

I can only supply an IP address or it fails with the following: 

```
❯ timoni build app .\haproxy\ --values .\haproxy\values.cue  --output json
5:15PM ERR build failed:
timoni.instance.config.targetServer: invalid value "cnn.com" (does not satisfy net.IP):
    .\templates\config.cue:102:19
    .\values.cue:5:19
``` 

But if I put the following into [cue playground](https://cuelang.org/play/#w=function&i=cue&f=eval&o=cue) it works (as expected) in either order:

```
import "net"

targetServer!: (net.IP | net.FQDN )
// targetServer: "foo.com"
targetServer: "1.1.1.1"
```

Any idea why the behavior would be different in timoni ?

**Comment by hansbogert** [comment]
is this reproducible? Or was this just the embedded cuelang in timoni which was misbehaving?

**Reply by binarytemple** [reply]
> is this reproducible? Or was this just the embedded cuelang in timoni which was misbehaving?

This was quite a while ago, but from the examples I gave it seemed to only behave that way in timoni

**Reply by hansbogert** [reply]
I think my question would've been clearer stated as, is this *still* reproducible?

**Reply by binarytemple** [reply]
Sorry Hans, I don’t have that setup now. My current client is using Cloudformation/ECS rather than Kubernetes, but it should be easy to reproduce, given the example above

**Comment by stefanprodan** [comment, marked as answer]
Tested this with Timoni v0.29.0 which embeds CUE v0.17.1 and the issue is fixed. Please upgrade!

**Reply by netfizzltd** [reply]
I’ll have another look at timoni then, helm sucks so bad it makes my eyes hurt. Thanks!
